src/metabase/util/i18n.clj

```diff
@@ -44,7 +48,9 @@
   "Locale string we should *use* for the current User (e.g. `tru` messages) -- `*user-locale*` if bound, otherwise the
   system locale as a string."
   []
-  (or *user-locale*
+  (or
+    *header-locale*
```

Comment:
The only place where we'll be sending the header AND the user will be logged in should be the sdk, in that case the header should win over the user setting